### PR TITLE
Player Coverart Priority system

### PR DESCRIPTION
**Pull request by defvs**
Allows the user to choose which coverart he wants in priority in the player.

I need your checks @Xerus2000  on this. The code might be a bit clunky but I haven't found that many solutions.

It works other than that. Works as expected, no side effects.

### CODE REVIEW

File: src/main/xerus/monstercat/api/Cache.kt [hunk 1]
```diff
@@ -38,10 +38,15 @@ object Cache: Refresher() {
 		return releases
 	}
 	
-	/** Gets all tracks by flatMapping all the tracks of all Releases */
+	/** Gets all tracks by flatMapping all the tracks of all Releases. Two tracks with the same id will only yield one (hashSet) */
 	suspend fun getTracks(): Collection<Track> =
 		getReleases().flatMap { it.tracks }.toHashSet()
 	
+	/** Gets all tracks by flatMapping all the tracks of all Releases. Will return all tracks, regardless of id duplicates.
+	 * This is very useful when you need all tracks for a release.*/
```
Comment (xeruf) [inline src/main/xerus/monstercat/api/Cache.kt]:
```suggestion
	 * This is very useful when you need a separate Track for each Release it is part of. */
```

File: src/main/xerus/monstercat/tabs/TabSettings.kt [hunk 2]
```diff
@@ -80,6 +83,20 @@ class TabSettings: VTab() {
 			Settings.PLAYERSEEKBARHEIGHT.bind(valueProperty() as ObservableValue<out Double>)
 		})
 		
+		val coverPriority = ComboBox(FXCollections.observableArrayList<String>())
+		onFx {
+			PriorityLists.values().forEach { priorities ->
+				coverPriority.items.add(PriorityLists.getString(priorities))
+				coverPriority.valueProperty().bindBidirectional(Settings.PLAYERARTPRIORITY, {
```
Comment (xeruf) [inline src/main/xerus/monstercat/tabs/TabSettings.kt]: I'm pretty sure I have a util function for this whole piece of code, please check out other ComboBoxes in the project

File: src/main/xerus/monstercat/tabs/TabSettings.kt [hunk 3]
```diff
@@ -224,4 +241,33 @@ class TabSettings: VTab() {
 		data class Feedback(val subject: String, val message: String)
 	}
 	
+	enum class PriorityLists(val priority: List<String>){
```
Comment (xeruf) [inline src/main/xerus/monstercat/tabs/TabSettings.kt]:
```suggestion
	enum class PriorityList(val priorities: List<String>){
```

**Review by xeruf — CHANGES_REQUESTED** [review]
LGTM, just a few minor comments

**Comment by defvs** [comment]
Fixes Xerus2000#60

File: src/main/xerus/monstercat/tabs/TabSettings.kt (hunk 3 above)
Comment (defvs) [reply src/main/xerus/monstercat/tabs/TabSettings.kt]: Sure thing.

File: src/main/xerus/monstercat/api/Cache.kt (hunk 1 above)
Comment (defvs) [reply src/main/xerus/monstercat/api/Cache.kt]: All clear for me.

File: src/main/xerus/monstercat/tabs/TabSettings.kt (hunk 3 above)
Comment (defvs) [reply src/main/xerus/monstercat/tabs/TabSettings.kt]: 1d87dd7eb67f094d676fa20417cc4dc1c39345f4

File: src/main/xerus/monstercat/api/Cache.kt (hunk 1 above)
Comment (defvs) [reply src/main/xerus/monstercat/api/Cache.kt]: cd4aa813bc62828d7cc18144b4bbb45f772e8223

File: src/main/xerus/monstercat/tabs/TabSettings.kt (hunk 2 above)
Comment (defvs) [reply src/main/xerus/monstercat/tabs/TabSettings.kt]: 2045bffad1e8fcbc3a72fe235f504ce706ac903f

**Comment by defvs** [comment]
Ready to merge;

**Pull request merged**